Public/Src/Engine/Dll/Distribution/WorkerService.cs

```diff
@@ -339,8 +344,7 @@ public void Exit(string failure, bool isUnexpected = false)
 
             if (!reportSuccess)
             {
-                // Only log the error, if this thread set exit response
-                Logger.Log.DistributionWorkerExitFailure(m_appLoggingContext, failure);
+                m_masterFailureMessage = failure;

```

Comment:
>failure [](start = 41, length = 7)

It might be a good idea to log a warning at the time that the failure message from the master comes in. That could be helpful for debugging other potential unexpected issues, like our eventual error not actually getting logged due to a different race condition or Exit() being called multiple times somehow